### PR TITLE
use NameIsDNSSubdomain validation from staging

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
> // TODO update all references to these functions to point to the apimachineryvalidation ones

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #67219

**Special notes for your reviewer**:
/cc seans3
@kubernetes/sig-apps-pr-reviews 
@kubernetes/sig-api-machinery-pr-reviews 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by nikhita** [comment]
@dixudx :+1: Since this updates references only, can the release note be removed?

/kind cleanup

File: pkg/apis/core/validation/validation.go [hunk 1]
```diff
@@ -195,23 +195,23 @@ type ValidateNameFunc apimachineryvalidation.ValidateNameFunc
...
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
-var ValidateReplicationControllerName = NameIsDNSSubdomain
+var ValidateReplicationControllerName = apimachineryvalidation.NameIsDNSSubdomain
 
 // ValidateServiceName can be used to check whether the given service name is valid.
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
-var ValidateServiceName = NameIsDNS1035Label
+var ValidateServiceName = apimachineryvalidation.NameIsDNS1035Label
```
Comment (liggitt) [inline pkg/apis/core/validation/validation.go]: if we're changing this, should we remove the function in this package?

Comment (dixudx) [reply pkg/apis/core/validation/validation.go]: Yeah. Right.

**Comment by dixudx** [comment]
ptal @liggitt

**Pull request merged**